### PR TITLE
Various fixes for training

### PR DESCRIPTION
**Pull request by michaelbenayoun**
# What does this PR do?

- Fixes the MPMD that happens when `logging_steps < num_step_in_epoch` at the end of the epoch.
- Fixes `has_write_access_to_repo`

### CODE REVIEW

File: optimum/neuron/utils/cache_utils.py [hunk 1]
```diff
@@ -127,7 +129,59 @@ def is_private_repo(repo_id: str) -> bool:
...
+        return True  # has write access, otherwise would have been 403 forbidden.
+    except HfHubHTTPError as e:
+        if e.response.status_code == 403:
+            return False
+        else:
+            raise ValueError(f"Cannot determine write access to {repo_id}") from e
+
+
+@cache_has_write_access
+def has_write_access_to_repo_without_api_action(repo_id: str) -> bool:
```
Comment (JingyaHuang) [inline optimum/neuron/utils/cache_utils.py:177]: where do you plan to use it?

**Review by JingyaHuang — APPROVED** [review]
Not familiar with the MPMD issue, the changes look good to me in general.

File: optimum/neuron/utils/cache_utils.py (hunk 1 above)
Comment (michaelbenayoun) [reply optimum/neuron/utils/cache_utils.py:177]: I actually do not plan to use it, it is more like if we need that in the future. We can remove that as well.

Comment (michaelbenayoun) [reply optimum/neuron/utils/cache_utils.py]: Removed it!

**Pull request merged**